### PR TITLE
[enhancement](memory) Support Jemalloc metrics and default allocator changed to Jemalloc 

### PR DESCRIPTION
**Pull request by xinyiZzz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Did not integrate Jemalloc heap profile, encountered some difficulties, will be added to action later.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.84 seconds
 load time: 444 seconds
 storage size: 17150793359 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221117224553_clickbench_pr_48333.html

File: be/src/util/mem_info.h [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include <string>
 
 #include "common/logging.h"
+#include "jemalloc/jemalloc.h"
```
Comment (adonis0147) [inline be/src/util/mem_info.h:28]: Need macro guard `USE_JEMALLOC` here?

Comment (xinyiZzz) [reply be/src/util/mem_info.h:29]: fix, which is more elegant

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**